lib/holiday_jp.rb

```diff
@@ -3,13 +3,34 @@
 require 'holiday_jp/holidays'
 
 module HolidayJp
-  def self.between(start, last)
-    Holidays.new.holidays.find_all do |date, _holiday|
-      start <= date && date <= last
-    end.map(&:last)
-  end
 
-  def self.holiday?(date)
-    Holidays.new.holidays[date]
+  class << self
+    def between(start, last)
+      Holidays.new.holidays.find_all do |date, _holiday|
+        start <= date && date <= last
+      end.map(&:last)
+    end
+
+    def holiday?(date)
+      Holidays.new.holidays[date]
+    end
+
+    def add_custom_holiday_sources(*files)
+      files.to_a.flatten.compact.uniq.each do |file|
+        next unless File.exist? file
+        add_custom_holiday_source file
+      end
+    end
+
+    def add_custom_holiday_source(source)
+      @custom_holiday_sources ||= []
+      @custom_holiday_sources << source
+    end
+
+    def default_holiday_sources
+      [File.expand_path('../../holidays.yml', __FILE__)].concat(@custom_holiday_sources.to_a).flatten.freeze
```

Comment:
Line is too long. [108/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.